src/main/windows/callsWidgetWindow.ts

```diff
@@ -443,53 +410,63 @@ export class CallsWidgetWindow {
                 };
             });
 
-            if (message.length > 0) {
-                view.sendToRenderer(DESKTOP_SOURCES_RESULT, message);
-            }
+            return message;
         }).catch((err) => {
             log.error('desktopCapturer.getSources failed', err);
 
             view.sendToRenderer(CALLS_ERROR, screenPermissionsErrMsg);
             this.win?.webContents.send(CALLS_ERROR, screenPermissionsErrMsg);
+
+            return [];
         });
     }
 
-    private handleCreateCallsWidgetWindow = async (viewId: string, msg: CallsJoinCallMessage) => {
+    private handleCreateCallsWidgetWindow = async (event: IpcMainInvokeEvent, msg: CallsJoinCallMessage) => {
         log.debug('createCallsWidgetWindow');
 
         // trying to join again the call we are already in should not be allowed.
         if (this.options?.callID === msg.callID) {
-            return;
+            return Promise.resolve();
         }
 
         // to switch from one call to another we need to wait for the existing
         // window to be fully closed.
         await this.close();
 
-        const currentView = ViewManager.getView(viewId);
+        const currentView = ViewManager.getViewByWebContentsId(event.sender.id);
         if (!currentView) {
             log.error('unable to create calls widget window: currentView is missing');
-            return;
+            return Promise.resolve();
         }
 
+        const promise = new Promise((resolve) => {
+            const connected = (ev: IpcMainEvent, incomingCallId: string) => {
+                log.debug('onJoinedCall', incomingCallId);
+
+                if (!this.isCallsWidget(ev.sender.id)) {
+                    log.debug('onJoinedCall', 'blocked on wrong webContentsId');
+                    return;
```

Comment:
This is one I didn't want to change the behaviour of quite yet until I had @streamer45 look at it.

Reply:
So the check is here now that we're merging the preload script, and yes in a sane environment this shouldn't happen. I can add a rejection so that it's properly rejection as long as that works for you :)